### DISCUSSION TITLE
Implementation with locale as second path segment (tenant-like)?

### DISCUSSION THREAD

**Discussion by shi-rudo** [amannn/next-intl#1503] [Q&A]
Hello!
unfortunately, I'm having a hard time getting next-intl to work tenant-like. 
My main problem is the implementation of something like this:

`example.com/country-one/:locale/page/...`
`example.com/country-two/:locale/page/...`

It does not have to be a dynamic path segment for the country segment.
/app/country-one/[locale]/page.tsx
/app/country-two/[locale]/page.tsx

My biggest problem is that next-intl seems to assume that the locale is always the first path segment. Is there any way I can change this behavior?

Thanks a lot!

**Comment by amannn** [comment]
Maybe [custom prefixes](https://next-intl-docs.vercel.app/docs/routing#locale-prefix-custom) could work for you?

About multi-tenancy, there’s a discussion here: https://github.com/amannn/next-intl/issues/1107
